File: iosApp/iosApp/data/mappers/media/PhotoMapper.swift
//
//  PhotoMapper.swift
//  iosApp
//
//  Created by nVoxel on 25.04.2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import TDLibKit
import TGDriveKit

class PhotoMapper {
    
    private let fileMapper: FileMapper
    
    init(fileMapper: FileMapper) {
        self.fileMapper = fileMapper
    }
    
    func toResponse(photo: Photo) -> CommonPhoto {
        return CommonPhoto(
            sizes: toResponse(sizes: photo.sizes)
        )
    }
    
    func toResponse(sizes: [PhotoSize]) -> [CommonPhotoSize] {
        return sizes.map { toResponse(size: $0) }
    }
    
    private func toResponse(size: PhotoSize) -> CommonPhotoSize {
        return CommonPhotoSize(
            type: size.type,
            photo: fileMapper.toResponse(file: size.photo),
            width: Int32(size.width),
            height: Int32(size.height)
        )
    }
}

File: source/data/mappers/src/androidMain/kotlin/com/voxeldev/tgdrive/mappers/media/PhotoMapper.kt
package com.voxeldev.tgdrive.mappers.media

import org.drinkless.td.libcore.telegram.TdApi.Photo
import org.drinkless.td.libcore.telegram.TdApi.PhotoSize
import com.voxeldev.tgdrive.main.api.media.Photo as CommonPhoto
import com.voxeldev.tgdrive.main.api.media.PhotoSize as CommonPhotoSize

/**
 * @author nvoxel
 */
class PhotoMapper(
    private val fileMapper: FileMapper,
) {

    fun toResponse(photo: Photo): CommonPhoto =
        CommonPhoto(sizes = toResponse(sizes = photo.sizes))

    fun toResponse(sizes: Array<PhotoSize>): List<CommonPhotoSize> =
        sizes.map { it.toResponse() }

    private fun PhotoSize.toResponse(): CommonPhotoSize =
        CommonPhotoSize(
            type = type,
            photo = fileMapper.toResponse(file = photo),
            width = width,
            height = height,
        )
}
